Beta National Guard Camp changes

91247686 opened the pull request:
* Cut bot numbers to about 1/3 of original. 
* - They are still numerous enough to be very dangerous, but they are now surmountable.
* Added more worthwhile loot to weapons lab.
* - Rail, plasma, fusion, emp and laser guns alongside mods  for the lasers.
* Altered bot names and descriptions to reflect their capabilities.
* Bots now drop their ammo.
* Bots now use milspec ammo, flamer bot uses napalm.
* Bots have a short delay on their attack, ~~and an audible warning (if one is close enough to hear it)~~. 
* - The delay serves as a representation of the time it takes for the bot to aim its weapon, with exact numbers acquired by testing the player's aim time with those weapons, stats, skills and arm/eye encumberance of 15/5.
* - This  serves to give the player a small chance to escape, rather than simply being killed instantly. 
* - ~~Delay only applies to players/NPCs, ensuring that bots remain effective against monsters.~~
* Made most doors able to be opened. Exceptions are large garage doors and those leading to armouries. 
* - These doors can still be cut open.
* New bot: W11H20B4 anti-materiel bot, armed with an integrated Barrett M107A1, it's a credible threat to even Heavy Power Armoured players.
* - I'm not quite sure what those letters and numbers mean, exactly. If anyone knows better, do tell and I'll change it.
* Added Tank Drone spawns. There will usually be two in the Camp's motor pool.
* Added warning signs around the Camp's perimeter. They read:
* - "Property of the United States National Guard.  Intruders will be shot on sight."

Dropped audible warning (well, technically it's still there, it just has a volume of 0).
Delay now applies to monsters as well, if I'm to use the aim time justification it needs to be consistent.

This post goes into a bit more detail: https://discourse.cataclysmdda.org/t/aggregates-mods/14741/20?u=aggregate

cainiaowu commented on data/mods/National_Guard_Camp/military.json:
```diff
@@ -2,8 +2,8 @@
     "type": "MONSTER",
-    "name": "W11B10",
-    "description": "Wraitheon (11B) Infantry 10 Level.  Part of Wraitheon's series of one-to-one auxiliaries designed to seamlessly integrate with more traditional forces.",
+    "name": "rifle bot",
```
Why not "W11B10 rifle bot"? "rifle bot" is way too generic.

91247686 replied:
Sure, I'll change it.

Night-Pryanik commented on data/mods/National_Guard_Camp/item_groups.json:
```diff
@@ -135,6 +137,29 @@
+      [ "hk_g80", 2 ],
+      [ "ftk93", 2 ],
+      [ "plasma_rifle", 2 ],
+	  [ "emp_gun", 5 ],
```
Indentation here and below.

91247686 replied:
Whoops, I'll fix that.

BorkBorkGoesTheCode commented on data/mods/National_Guard_Camp/item_groups.json:
```diff
@@ -56,13 +56,15 @@
+      { "group": "guns_milspec", "prob": 20 },
+      { "group": "mags_milspec", "prob": 20 },
+      { "group": "energy_weapon_armory", "prob": 17 },
+      { "group": "hightec_guns", "prob": 3 },
```
I would recommend changing this to "guns_energy" itemgroup.

91247686 replied:
The energy_weapon_armory group is there because it has laser weapon mods in it. But sure, I'll just add those to the hightec group.

kevingranade commented:
> Cut bot numbers to about 1/3 of original.

Your justification for this in https://discourse.cataclysmdda.org/t/new-mega-map-feature-from-acidia-national-guard-camp/13826/8, "I don’t think anyone is even slightly likely to survive them without at least basic powered armour." is pretty much working as intended, we *want* places that require power armor to assault.
> Added more worthwhile loot to weapons lab.
> Altered bot names and descriptions to reflect their capabilities.

Both of these are good.
> Bots now drop their weapons/ammo.

Dropping the ammo is fine, but they aren't weilding person-usable weapons.
> Bots now use milspec ammo, flamer bot uses napalm.

Cool.
> Bots have a short delay on their attack, and an audible warning (if one is close enough to hear it).

Not ok, these are "shoot-to-kill" type robots intentionally.  If you're concerned about players being caught off guard, feel to sprinkle some warning signs around the perimiter.

> Made most doors able to be opened.

Cool.

91247686 commented:
My justification for the bot cut is that there were just too many of them. Wading through 300+ *exceedingly* dangerous foes is far from fun, and this being a game I think fun is more important than most other things. Fighting through ~100 of them *is* a fun and tense experience, with their numbers high enough to be a great challenge, but not so high that one can expect to fight more than ~5 at a time.
Surviving the camp without PA is still quite unlikely, especially if I'm to remove the delay.

Actually using basic or heavy PA (with light PA being only effective enough to survive 2-3 bullets), renders the camp a non-issue, with only flamer and grenade bots capable of actually harming the player, and even then, they'll likely run out of ammo before killing the player. I'm fine with places requiring PA to assault, but I wouldn't call a binary instant death/non-issue location good design.
The solution to this is nerfing PA,  nerfing the area or making the area more dangerous. Which I'm fine with, as long as there are clear warnings that this is not a place you want to go. Though, the problem with increasing danger is that nothing poses a threat to PA save tank drones and elite grenadiers.
Hm, I could add a few tank drones to the camp. Worth a try.

But adding tank drones or similar won't make the bots capable of killing PA'd players.
The question becomes: keep bot numbers high even though they're a non-issue for PA'd players, or lower them so a skilled player has a *slight* chance of getting something out of the camp, even without PA.

As for the delay, I don't consider instantly killing the player to be acceptable (certainly not as a warning), unless it's a mistake the player *chooses* to make. Attempting to disarm a mine without the skills required, for example.
However, if you insist, I'll remove it. And I'll see about adding warning signs regardless.

I'll remove the weapon drops.

91247686 commented:
Closed while I do some more work.

BorkBorkGoesTheCode commented:
@91247686 the item_groups.json file has an extra ] at the end of the file.

BorkBorkGoesTheCode commented:
@91247686 that broke the file.

91247686 commented:
Fixed it.
Probably going to be a while before I have time to do more work on this.

BorkBorkGoesTheCode commented:
Ok.

BorkBorkGoesTheCode commented:
@91247686 any updates?

91247686 commented:
I'll have more time in a few days, I'll work on it then.

91247686 commented:
I've now done some more work.

* Added Tank Drone spawns. Two will usually spawn in the motor pool.
* Changed the sniper bot's weapon to the M2010 ESR, which allows it to harm Basic PA'd players, though it still can't scratch Heavy PA. Also increased range and added more spawns.
* Added new bot: W11H20B4 antimaterial bot, armed with an integrated Barrett M107A1, it is quite capable of killing even Heavy PA'd players. It's pretty rare, though.
* - I'm not quite sure what those letters and numbers mean, exactly. If anyone knows better, do tell and I'll change it.
* Changed delay times to serve as a representation of the time it takes for the bot to aim its weapon, with exact numbers acquired by testing the player's aim time with those weapons, stats, skills and arm/eye encumberance of 15/5.
* - This makes firefights rather more fair, with bots and players both having to take time to aim. Though, unlike players, bots seem to land every shot regardless of their skills. Bots always have to aim from scratch after a shot, though, and I'm not sure if it's possible to change that.
* Added warning signs around the Camp's perimeter. They read: "Property of the United States National Guard.  Intruders will be shot on sight."
* Altered bot colours, allowing players to tell them apart at a glance.
* Increased likelihood of special (non rifle) bots spawning.
* Altered bot descriptions to state that their weaponry is integrated, implying that they cannot be removed and used by the player.
* Dropped audible warnings completely.

Ran into an odd issue where monstergroups almost always spawned the default monster, rather than one from the group. Worked around it by changing the defaults to the bots I wanted to spawn.

Any objections (with arguments) or suggestions?

BorkBorkGoesTheCode commented:
@91247686 acidia layed out the robot names here: https://github.com/CleverRaven/Cataclysm-DDA/pull/19257#issuecomment-262860333

91247686 commented:
Okay, looking at that tells me it's about right, but what was "11H" now falls under "11B" so I'll change that.

kevingranade commented:
The volume of changes is a massive impediment to review, I predict this will continue to sit at the bottom of my review queue indefinitely because I'm unlikely to ever feel like reading through thousands of lines of json changes. I suspect the majority of changes, and likely the most important ones, can be made with a much smaller number of changes.

You never changed my mind about spawn volumes, basically you never made an argument for it at all. I don't know how many bots spawn in one of these areas offhand, it may well be too many, but you haven't bothered to say how many, so I'm just reflexively saying no.  I might eventually get around to booting up the game and doing that legwork myself, but I can't tell you when I'll do that.  

I'd also have to take a look at the delay times, It very much looks like, "I'm agreeing to not make this change, but I'm making as much of it as I think I can get away with".  These are literally killer military robots, nerfing them to be "fair" compared to a player is kind of nonsense, and like with changing the bot numbers, there's no argument for why it needs to happen.

91247686 commented:
What you're saying is absurd. The delay is both realistic and *fair*. Unless your idea of killer military robots involves them magically turning to face targets and aiming at said targets *instantly,* I really don't understand why you have such a problem with it. And again, similarly lethal foes the Tank Drone and Chicken Walker have a similar system *specifically* (if the comments are to be believed) to give the player a chance to escape, so the delay is consistent with existing hyper-lethal enemies. Though I'd say the Camp bots are far more dangerous than the TD or CW, "instantly kills you" as opposed to "instantly kills you after giving you a chance to get to cover". If you don't consider *fairness* to be a very important part of gameplay, then I don't know what to say.

> "I'd also have to take a look at the delay times, It very much looks like, "I'm agreeing to not make this change, but I'm making as much of it as I think I can get away with"."

Nowhere did I say or imply that I was agreeing to not make the change, only that I'd remove it if you insisted, which makes this statement completely absurd.

I did in fact state that there are 300+ bots in my reply to your first post here, maybe you should've bothered to read it before claiming I'd never bothered to say. 

My argument is *fun.* As I stated above, fighting through the reduced bots was (before repetition left me inured) fun, whereas fighting through the 300 - 400 bots there normally was a horrible slog through seemingly never-ending, *exceedingly* deadly foes, where the slightest mistake was punished with *instant death* unless one was wearing PA, in which case it was a complete (too easy to be fun) cakewalk. 
Oh, and one of those mistakes? Not using a laser rifle with very high skills from beyond the bot's max range. Hardly good design, restricting players to a single playstyle in a game that normally encourages a diverse array thereof.

Though, if one is wearing heavy PA, it's still largely a cakewalk (save for two foes that can one- or two-shot such a player). But that has to do with PA being very overpowered.

If, for some reason, you don't see fun as valid, then I *really* don't know what to say to you, except that perhaps you shouldn't be running the development of a videogame, or that I would rather punch a dozen tank drones to death before setting foot anywhere *near* another NG Camp, without these (or similar) changes. Though I rather doubt you'd take such a statement seriously (er, I'm referring to the latter statement, here. You obviously wouldn't take the former seriously).

As for the number of line changes, I *would* say that I'll see about reducing them, but interacting with you has left me bereft of motivation. Maybe at some unknown point in the future.

I've made my arguments, how about you make yours?

kevingranade commented:
I *do* insist you remove the aim time and the laser lock, those are specifically there to slow down attackers the player will run across at random.  This is not necessary for attackers in a clearly marked killzone.  This is non-negotiable, you're either willing to do that, or this PR is dead.

To be clear, no I don't care about *fairness*, the game isn't remotely *fair* in any direction, it's wildly assymetrical by design in that you're expected to be consistently outmatched in some ways and consistently overpowered in others, it's up to you to make sure you exercise your strengths and hide your weaknesses.  Trying to outshoot a group of milspec robots is something I would characterize as a *bad idea*, and I have no interest in nerfing challenges so that players that have bad ideas can succeed.
